expression/constant_propagation.go

```diff
@@ -158,6 +150,49 @@ func (s *propagateConstantSolver) validPropagateCond(cond Expression, funNameMap
 	return nil, nil
 }
 
+// tryToReplaceCond aims to replace all occurances of column 'src' and try to replace it with 'tgt' in 'cond'
+// It returns
+//  bool: if a replacement happened
+//  bool: if 'cond' contains non-deterministic expression
+//  Expression: the replaced expression, or original 'cond' if the replacement is not happened
+func (s *propagateConstantSolver) tryToReplaceCond(src *Column, tgt *Column, cond Expression) (bool, bool, Expression) {
+	replaced := false
+	var r *ScalarFunction
+	if sf, ok := cond.(*ScalarFunction); ok {
+		if _, ok := unFoldableFunctions[sf.FuncName.L]; ok {
+			return false, true, cond
+		}
+		if _, isEq := eqFuncNameMap[sf.FuncName.L]; isEq {
+			return false, false, cond
+		}
+		for idx, expr := range sf.GetArgs() {
+			if src.Equal(nil, expr) {
+				replaced = true
+				if r == nil {
+					r = sf.Clone().(*ScalarFunction)
+				}
+				r.GetArgs()[idx] = tgt
```

Comment:
You'd better call `NewFunction`. Don't set `scalar function`'s arg.

Reply:
OK~